lib/shift/api/core/middleware/custom_headers.rb

```diff
@@ -6,9 +6,9 @@ module Middleware
         # Faraday middleware to add extra headers to the request
         #
         class CustomHeaders < ::Faraday::Middleware
-          def initialize(app, headers:)
+          def initialize(app, options = {})
             self.app = app
-            self.headers = headers
+            self.headers = options.fetch(:headers)
           end
```

Comment:
Need to test this on Ruby 2.7 as well, to ensure it doesn't cause a regression there. If it does and it's not easy to fix then we can just do a major version bump on the gem

Reply:
For reference for anyone wondering, we're having issues with Ruby 3 and the JsonApiClient dependency, as that gem doesn't support the kwarg changes in Ruby 3.

Reply:
Yeah it's due to JSON API not passing the `kwargs` properly into this faraday builder https://github.com/JsonApiClient/json_api_client/blob/4277a862edf2f80a168c2c7e0206325a9eda5a4a/lib/json_api_client/connection.rb#L27